src/main/java/com/exasol/validation/GitRepositoryValidator.java

```diff
@@ -33,39 +36,60 @@ public void validate(final String branch) {
-        validateVersionFormat(newVersion);
-        final Optional<String> latestReleaseTag = this.repository.getLatestTag();
-        if (latestReleaseTag.isPresent()) {
-            validateNewVersionWithPreviousTag(newVersion, latestReleaseTag.get());
+        final boolean correctVersionFormat = validateVersionFormat(newVersion);
+        boolean newReleaseTagIsValid = false;
+        if (correctVersionFormat) {
+            newReleaseTagIsValid = validateIfNewReleaseTagValid(newVersion);
         }
+        return correctVersionFormat && newReleaseTagIsValid;
     }
 
-    // [impl->dsn~validate-release-version-format~1]
-    private void validateVersionFormat(final String version) {
-        if (!version.matches(VERSION_REGEX)) {
-            throw new IllegalArgumentException(
-                    "E-RR-VAL-3: A version or tag found in this repository has invalid format. "
+    private boolean validateVersionFormat(final String version) {
+        if (version.matches(VERSION_REGEX)) {
+            this.validationReport.addSuccessfulValidation("Version format.");
+            return true;
+        } else {
+            this.validationReport.addFailedValidations("E-RR-VAL-3",
+                    "A version or tag found in this repository has invalid format. "
                             + "The valid format is: <major>.<minor>.<fix>. "
                             + "Please, refer to the user guide to check requirements.");
+            return false;
+        }
+    }
+
+    private boolean validateIfNewReleaseTagValid(final String newVersion) {
+        final Optional<String> latestReleaseTag = this.repository.getLatestTag();
+        if (latestReleaseTag.isPresent()) {
+            return validateNewVersionWithPreviousTag(newVersion, latestReleaseTag.get());
+        } else {
+            this.validationReport.addSuccessfulValidation("A new tag. This is the first release.");
+            return true;
         }
     }
+    // [impl->dsn~validate-release-version-format~1]
 
     // [impl->dsn~validate-release-version-increased-correctly~1]
-    private void validateNewVersionWithPreviousTag(final String newTag, final String latestTag) {
+    private boolean validateNewVersionWithPreviousTag(final String newTag, final String latestTag) {
         final Set<String> possibleVersions = getPossibleVersions(latestTag);
-        if (!possibleVersions.contains(newTag)) {
-            throw new IllegalArgumentException(
-                    "E-RR-VAL-4: A new version does not fit the versioning rules. Possible versions for the release are: "
+        if (possibleVersions.contains(newTag)) {
+            this.validationReport.addSuccessfulValidation("A new tag.");
+            return true;
+        } else {
+            this.validationReport.addFailedValidations("E-RR-VAL-4",
+                    "A new version does not fit the versioning rules. Possible versions for the release are: "
                             + possibleVersions.toString());
```

Comment:
Have you checked that `possibleVersions.toString()` renders as expected. Not that it renders to Set@43345435 or similar

Reply:
I removed that test after I added the error codes. You're right, it's better to have it

Reply:
So you checked it? I just don't belive it produces the correct output...

Reply:
Yes, I've added a test, you can find it in the pull request

Reply:
:+1: